code/datums/martial/worldshaker.dm

```diff
@@ -0,0 +1,563 @@
+
+	var/obj/effect/temp_visual/decoy/D = new /obj/effect/temp_visual/decoy(user.loc,user)
+	animate(D, alpha = 0, color = "#000000", transform = matrix()*2, time = 0.3 SECONDS)
+	animate(user, time = (heavy ? 0.4 : 0.2)SECONDS, pixel_y = 20)//we up in the air
+	addtimer(CALLBACK(src, PROC_REF(reset_pixel), user), 1.5 SECONDS, TIMER_UNIQUE | TIMER_OVERRIDE)//in case something happens, we don't permanently float
+	playsound(user, 'sound/effects/gravhit.ogg', 15)
+	playsound(user, 'sound/effects/dodge.ogg', 15, TRUE)
+
+/datum/martial_art/worldshaker/proc/leap_end(mob/living/carbon/human/user)
+	user.SetImmobilized(0 SECONDS, ignore_canstun = TRUE)
+	leaping = FALSE
+	var/range = LEAP_RADIUS
+	if(heavy)
+		range++
+		
+	for(var/mob/living/L in range(range,user))
+		if(L == user)
+			continue
+		stagger(L)
+		var/damage = heavy ? 25 : 15 //chunky boy does more damage
+
+		if(L.loc == user.loc)
+			to_chat(L, span_userdanger("[user] lands directly ontop of you, crushing you beneath their immense weight!"))
+			damage *= 2//for the love of god, don't get landed on
+			L.adjustStaminaLoss(damage)
+
+		L.apply_damage(damage, BRUTE, wound_bonus = 10, bare_wound_bonus = 20)
+		push_away(user, L)
+	for(var/obj/item/I in range(range, user))
+		push_away(user, I)
+
+	animate(user, time = 0.1 SECONDS, pixel_y = 0)
+	playsound(user, 'sound/effects/gravhit.ogg', 20, TRUE)
+	playsound(user, 'sound/effects/explosion_distant.ogg', 200, FALSE, WARNING_RANGE)
+	var/atom/movable/gravity_lens/shockwave = new(get_turf(user))
+	shockwave.transform *= 0.1 //basically invisible
+	shockwave.pixel_x = -240
+	shockwave.pixel_y = -240
+	animate(shockwave, alpha = 0, transform = matrix().Scale(range/3), time = 1 + (range/10))
+	QDEL_IN(shockwave, 2 + (range/10))
+
+/datum/martial_art/worldshaker/proc/reset_pixel(mob/living/user)//in case something happens, we don't permanently float
+	animate(user, time = 0.1 SECONDS, pixel_y = 0)
+
+/datum/martial_art/worldshaker/handle_throw(atom/hit_atom, mob/living/carbon/human/A)
+	if(leaping)
+		return TRUE
+	return ..()
+/*---------------------------------------------------------------
+	end of leap section
+---------------------------------------------------------------*/
+/*---------------------------------------------------------------
+	start of grapple section
+---------------------------------------------------------------*/
+/datum/martial_art/worldshaker/proc/grab(mob/living/user, mob/living/target, damage)//proc the moves will use for damage dealing
+	stagger(target)
+	var/obj/item/bodypart/limb_to_hit = target.get_bodypart(user.zone_selected)
+	var/armor = target.run_armor_check(limb_to_hit, MELEE, armour_penetration = 35)
+	target.apply_damage(damage, BRUTE, limb_to_hit, armor, wound_bonus=CANT_WOUND)
+	
+/datum/martial_art/worldshaker/proc/drop(mob/living/target)//proc for clearing the thrown list, mostly so the lob proc doesnt get triggered when it shouldn't
+	for(var/atom/movable/K in thrown)
+		thrown.Remove(K)
```

Comment:
when i said vars i meant everywhere cause it gets very confusing sometimes

Reply:
yeah, this entire ability is copy pasted from buster arm, so a bunch of variables are kinda wack
i've renamed pretty much every variable now